src/tmx/Messages/include/SensorDetectedObject.h

```diff
@@ -28,23 +28,115 @@ namespace tmx
-            
-            // Classification of detected object
-            std::string type = "";
+            //Flag to indicate whether sensor detected object is simulated.
+            std_attribute(this->msg, bool, ISSimulated, false,);
+            // Classification of detected object.
+            std_attribute(this->msg, std::string, Type, "",);
             // Confidence of type classification
-            double confidence = 0.0;
-            // Unique indentifier of sensor reporting detection
-            std::string sensorId = "";
-            // String describing projection used to convert cartesian data to WGS84 data 
-            std::string projString = "";
-            // Unique identifier of detected object
-            int objectId = 0;
+            std_attribute(this->msg, double, Confidence, 0.0,);
+            // Unique indentifier of sensor reporting detection.
+            std_attribute(this->msg, std::string, SensorId, "", );
+            // String describing projection used to convert cartesian data to WGS84 data.
+            std_attribute(this->msg, std::string, ProjString, "", );
+            // Unique identifier of detected object.
+            std_attribute(this->msg, int, ObjectId, 0, );
+
             // Cartesian positiion of object. Assumed to be ENU coordinate frame.
-            tmx::utils::Point position = tmx::utils::Point();
+            typedef struct Position{
+                double x,  y, z;
+                Position(){};
+                Position(double x, double y, double z):x(x),y(y),z(z){};
+                static message_tree_type to_tree(const Position& pos){
+                    message_tree_type tree;
+                    tree.put("x", pos.x);
+                    tree.put("y", pos.y);
+                    tree.put("z", pos.z);
+                    return tree;
+                }
+                static Position from_tree(const message_tree_type& tree){
+                    Position pos;
+                    pos.x = tree.get<double>("x");
+                    pos.y = tree.get<double>("y");
+                    pos.z = tree.get<double>("z");
+                    return pos;
+                }
+            } Position;            
+            object_attribute(Position, Position);
+
+            typedef struct Covariance{
+                std::string value;                
+                Covariance(){};
+                Covariance( std::string value):value(value){};
+                static message_tree_type to_tree(const Covariance& cov){
+                    message_tree_type tree;
+                    tree.put("", cov.value);
+                    return tree;
+                }
+                static Covariance from_tree(const message_tree_type& tree){
+                    Covariance cov;
+                    cov.value = tree.get<std::string>("");
```

Comment:
These will be double values.

Reply:
https://usdot-carma.atlassian.net/wiki/spaces/CRMSIM/pages/2563899417/Detected+Objects+Specification